internal/config/config.go

```diff
@@ -33,20 +35,27 @@ type dbConfig struct {
 }
 
 type svcConfig struct {
-	Address              string   `json:"address,omitempty"`
-	AgentEndpointAddress string   `json:"agentEndpointAddress,omitempty"`
-	AgentGrpcAddress     string   `json:"agentGrpcAddress,omitempty"`
-	CertStore            string   `json:"cert,omitempty"`
-	BaseUrl              string   `json:"baseUrl,omitempty"`
-	BaseAgentEndpointUrl string   `json:"baseAgentEndpointUrl,omitempty"`
-	BaseAgentGrpcUrl     string   `json:"baseAgentGrpcUrl,omitempty"`
-	BaseUIUrl            string   `json:"baseUIUrl,omitempty"`
-	CaCertFile           string   `json:"caCertFile,omitempty"`
-	CaKeyFile            string   `json:"caKeyFile,omitempty"`
-	SrvCertFile          string   `json:"srvCertFile,omitempty"`
-	SrvKeyFile           string   `json:"srvKeyFile,omitempty"`
-	AltNames             []string `json:"altNames,omitempty"`
-	LogLevel             string   `json:"logLevel,omitempty"`
+	Address               string   `json:"address,omitempty"`
+	AgentEndpointAddress  string   `json:"agentEndpointAddress,omitempty"`
+	AgentGrpcAddress      string   `json:"agentGrpcAddress,omitempty"`
+	CertStore             string   `json:"cert,omitempty"`
+	BaseUrl               string   `json:"baseUrl,omitempty"`
+	BaseAgentEndpointUrl  string   `json:"baseAgentEndpointUrl,omitempty"`
+	BaseAgentGrpcUrl      string   `json:"baseAgentGrpcUrl,omitempty"`
+	BaseUIUrl             string   `json:"baseUIUrl,omitempty"`
+	CaCertFile            string   `json:"caCertFile,omitempty"`
+	CaKeyFile             string   `json:"caKeyFile,omitempty"`
+	SrvCertFile           string   `json:"srvCertFile,omitempty"`
+	SrvKeyFile            string   `json:"srvKeyFile,omitempty"`
+	AltNames              []string `json:"altNames,omitempty"`
+	LogLevel              string   `json:"logLevel,omitempty"`
+	HttpReadTimeout       Duration `json:"httpReadTimeout,omitempty"`
+	HttpReadHeaderTimeout Duration `json:"httpReadHeaderTimeout,omitempty"`
+	HttpWriteTimeout      Duration `json:"httpWriteTimeout,omitempty"`
+	HttpMaxNumHeaders     int      `json:"httpMaxNumHeaders,omitempty"`
+	HttpMaxHeaderBytes    int      `json:"httpMaxHeaderBytes,omitempty"`
+	HttpMaxUrlLength      int      `json:"httpMaxUrlLength,omitempty"`
+	HttpMaxRequestSize    int      `json:"httpMaxRequestSize,omitempty"`
```

Comment:
It'd be awesome to have a bit more structure to the svcConfig and I was wondering whether user- and agent-facing servers would need different timeouts, but that can well be in a follow-up PR.

Reply:
Sure, let's discuss in a folluw-up